File: backend/app/api/v1/endpoints/reports.py
"""Reporting endpoints for administrative dashboards."""

from fastapi import APIRouter, Depends, Query

from app.api.deps import get_current_admin, get_report_service
from app.schemas.report import ReportOverview, ReportActivity
from app.services.reports import ReportService

router = APIRouter()


@router.get("/overview", response_model=ReportOverview)
def get_report_overview(
    days: int | None = Query(None, ge=1, le=365, description="统计时间范围（天）"),
    service: ReportService = Depends(get_report_service),
    current_admin=Depends(get_current_admin),
) -> ReportOverview:
    data = service.overview(days=days)
    return ReportOverview(**data)


@router.get("/activity/{activity_id}", response_model=ReportActivity)
def get_activity_report(
    activity_id: int,
    days: int | None = Query(None, ge=1, le=365, description="统计时间范围（天）"),
    service: ReportService = Depends(get_report_service),
    current_admin=Depends(get_current_admin),
) -> ReportActivity:
    data = service.activity_report(activity_id, days=days)
    return ReportActivity(**data)

File: frontend/admin/src/services/reports.ts
import http from './http'

export async function getOverview(days?: number) {
  const params: any = {}
  if (days) params.days = days
  const resp = await http.get('/reports/overview', { params })
  return resp.data
}

export async function getActivityReport(activityId: number, days?: number) {
  const params: any = {}
  if (days) params.days = days
  const resp = await http.get(`/reports/activity/${activityId}`, { params })
  return resp.data
}
